### ISSUE TITLE
Use Redox?

### ISSUE THREAD

**Issue by stevenvachon** [NodeOS/NodeOS#429]
It's written in Rust. It might be the future.

**Comment by piranna** [comment]
I know about Redox since some time ago, and I love its concept. Can you explain your idea in more detail?

**Comment by stevenvachon** [comment]
I can't explain further, heh. I know next to nothing about the internals of Linux and Unix. I just think that Rust is changing things.

**Comment by dzek69** [comment]
https://transitiontech.ca/random/RIIR couldn't be more accurate

(ok, it could, because it's just an advertisement of competetive solution, not a rewrite proposal)

still incredibly funny to find an unusual project, entering Issues to get know about current problems and limitations of the project, just to find Rust-stuff listed as first issue.

**Comment by piranna** [comment]
Yes, I know both Rust and Redox and I'm very interested on both of them, just only they are different beasts from NodeOS target purposse, although some ideas could be ported. If only Redox kernel were a standalone project...

Note: sorry to everybody for the radio silence during the last years, I have been really busy due to real life work and freelance jobs, and the few time I had was mostly with other small projects related with work too. Hope someday soon to have some time to take back NodeOS and polish it a bit, but must to say maybe with Rust and WebAssembly, or not! :-D

**Comment by erlend-sh** [comment]
At this point, a Redox + [Deno](https://deno.land/v1) operating system seems mighty appealing.

**Comment by piranna** [comment]
> At this point, a Redox + [Deno](https://deno.land/v1) operating system seems mighty appealing.

Benchmarks of Deno compared to Node.js are not really good at this moment, but I honestly love Rust as a language. Yes, I though about Redox too as a base layer for NodeOS, but didn't considered about combining it with Deno, although being both of them written in Rust could make it a good mix... but if going that path, maybe it would be better to combine Redox with Wasmer to have a system similar to what Nebulet was trying to achieve.

In any case, I'm open for disussions about this kind of topics, pull-requests, and to welcome others to became maintainers of NodeOS itself. I think it could be time to transfer the keys to others again...

**Comment by erlend-sh** [comment]
> maybe it would be better to combine Redox with Wasmer

Indeed. I [floated that](https://gitlab.redox-os.org/redox-os/redox/issues/1213) a while ago. Maybe @Hedzer has more to say about it now.
